pkg/cmd/server/kubernetes/node_config.go

```diff
@@ -291,3 +299,69 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig) (*NodeConfig, error
 
 	return config, nil
 }
+
+func buildKubeProxyConfig(options configapi.NodeConfig) (*kproxy.ProxyServerConfig, error) {
+	// get default config
+	proxyconfig := kproxy.NewProxyConfig()
+
+	// BindAddress - Override default bind address from our config
+	addr := options.ServingInfo.BindAddress
+	host, _, err := net.SplitHostPort(addr)
+	if err != nil {
+		return nil, fmt.Errorf("The provided value to bind to must be an ip:port %q", addr)
+	}
+	ip := net.ParseIP(host)
+	if ip == nil {
+		return nil, fmt.Errorf("The provided value to bind to must be an ip:port: %q", addr)
+	}
+	proxyconfig.BindAddress = ip
+
+	// HealthzPort, HealthzBindAddress - disable
+	proxyconfig.HealthzPort = 0
+	proxyconfig.HealthzBindAddress = nil
+
+	// OOMScoreAdj, ResourceContainer - clear, we don't run in a container
+	proxyconfig.OOMScoreAdj = 0
+	proxyconfig.ResourceContainer = ""
+
+	// use the same client as the node
+	proxyconfig.Master = ""
+	proxyconfig.Kubeconfig = options.MasterKubeConfig
+
+	// PortRange, use default
+	// HostnameOverride, use default
+
+	// ProxyMode, set to iptables
+	proxyconfig.ProxyMode = "iptables"
```

Comment:
This line has no effect, because the call to cmdflags.Resolve() below will overwrite it with "". (Maybe this is a bug in kubernetes/cmd/kube-proxy/app/server.go and `ProxyMode` there should default to `s.ProxyMode` rather than `""`?)